test/vm-upload

```diff
@@ -14,55 +17,50 @@ IMAGES = os.path.join(BASE, "images")
 DATA = os.path.join(os.environ.get("TEST_DATA", BASE), "images")
 
 def upload(store, source):
-    name = os.path.basename(source)
 
     # Compress the image
     compressed = source + ".xz"
-    with open(compressed, "w") as fp:
-        xz = [ "xz", "--force", "--keep", "--verbose", "--stdout", "--threads=0", source ]
-        subprocess.check_call(xz, stdout=fp)
+    if not os.path.exists(compressed):
+        partial = compressed + ".partial"
+        with open(partial, "w") as fp:
+            xz = [ "xz", "--force", "--keep", "--verbose", "--stdout", "--threads=0", source ]
+            subprocess.check_call(xz, stdout=fp)
+        os.rename(partial, compressed)
 
-    # And upload the compressed version
-    dest = os.path.join(store, name + ".xz")
-    subprocess.check_call(["rsync", "--progress", compressed, dest])
-
-    # Remove the compressed version
-    os.unlink(compressed)
-
-def prune(store):
-    # Get a list of all currently in use files
-    current = []
-    for filename in os.listdir(IMAGES):
-        path = os.path.join(IMAGES, filename)
-        if os.path.islink(path):
-            current.append(os.readlink(path))
-
-    # Get remote list of files order than the given date
-    (host, unused, path) = store.partition(":")
-    cmd = ["ssh", host, "--", "find", path, "-name", "'*.xz'", "-mtime", "+{0}".format(testinfra.IMAGE_EXPIRE), "-print0"]
-    output = subprocess.check_output(cmd)
 
-    remove = []
-    for path in output.strip('\0').split('\0'):
-        for name in current:
-            if name in path:
-                break
-        else:
-            remove.append(path)
-
-    cmd = ["ssh", host, "--", "rm", "-vf" ] + remove
-    subprocess.check_call(cmd)
+    # And upload the compressed version
+    ca = os.path.join(BASE, "common", "ca.pem")
+    (address, delim, port) = UPLOAD.partition(":")
```

Comment:
Should this use store? Otherwise that option might need to be removed. I think would be nice to keep it though now it probably needs to always be an ip.